### ISSUE TITLE
OpenSUSE Leap 42.1 and Tumbleweed - dotnet crash "failed to initialize" hresult 0x80131500

### ISSUE THREAD

**Issue by wpostma** [dotnet/core#207]
Following the instructions provided on the get .net site, with the ICU and libunwind installed, dotnet is crashing:

wpostma@warrenspc:~> dotnet version
Failed to initialize CoreCLR, HRESULT: 0x80131500

Tried this on both Tumbleweed (a very bleeding edge) and stable OpenSuse Leap 42.1.
Both are x86_64 architecture.

I will gather more information if I can but I can't find the documentation on how to generate a more detailed crash log.

Attached strace of dotnet version on Tumbleweed. It does appear it's not finding libicu.so.53.  

[dotnet_version_strace.txt](https://github.com/dotnet/core/files/351159/dotnet_version_strace.txt)

My distro only provides these libicuuc files:
/usr/lib64/libicuuc.so
/usr/lib64/libicuuc.so.57.1
/usr/lib64/libicuuc.so.57_1

I am not sure what the thinking was in linking directly to ONLY /usr/lib64/libicuuc.so.53.1
If I fake it by symlinking the .53 filename to the existing .57 libs, I get a different error:

dotnet: symbol lookup error: /opt/dotnet/shared/Microsoft.NETCore.App/1.0.0/System.Globalization.Native.so: undefined symbol: uloc_getDefault_53_1

In my arrogant outsider view, I wonder why something as important as the dotnet executable uses a SYSTEM  shared library like libicu when you can never guarantee these libraries will exist. Why not ship them as part of the dotnet binaries?

**Comment by wpostma** [comment]
I'm going to try to build from sources on OpenSuse Tumbleweed. Haven't found a readme. 

But I did find a repo that is necessary in order to get some of the rare dependencies like LTTNG-UST-DEVEL.

http://software.opensuse.org/download.html?project=home%3Admacvicar%3Adotnet&package=lttng-ust

OpenSuse dependency/tool fetch "zypper install lttng-ust-devel" will work after adding the above repo.
lttng-ust-devel is some trace tool that .net core depends on.

If this issue should be logged somewhere else let me know.

**Comment by wpostma** [comment]
I have asked this question and will self answer it on stack overflow if I can figure out what the recommended way to diagnose startup crashes is going to be on dotnet tool crashes, on Linux.

Since Linux dependency hell will be a common theme now that Linux support has shipped, this is going to be fun.

http://stackoverflow.com/questions/38235855/when-dotnet-tool-wont-start-up-failed-to-initialize-what-environment-variable

**Comment by svick** [comment]
I believe the problem is that the only supported version of OpenSUSE is 13.2.

**Comment by wpostma** [comment]
I also tried installing ICU 53  libs (built from source) and I still get the dotnet: symbol lookup error: /opt/dotnet/shared/Microsoft.NETCore.App/1.0.0/System.Globalization.Native.so: undefined symbol: uloc_getDefault_53_1

I believe it's dying pretty early in startup.  I'll keep updating this issue if I figure out how to get it to work.

I've tried building dotnet coreclr, dotnet corefx, and dotnet cli from sources, and have not had complete success.  Of course part of my trouble is that it's not easy to build the preview3 bits from sources when the documentation is incomplete.

I'm studying a working system (centos 7) and will use that working system to figure out how to get OpenSuse Tumbleweed to work.   

It seems OBVIOUS to me that the place to target a brand new platform like Dotnet Core on linux is on a bleeding edge rolling release distribution, which will become the future state of ALL linux desktops. For example, OpenSUSE Tumbleweed is using Linux 4.6.2, which I hear will break dotnet if you upgrade the kernel on a working dotnet install, dotnet's coreclr internals stop working.
